### PR TITLE
fix: Sort deployments in descending order and fix APIM arm template

### PR DESCRIPTION
**Pull request by tbarlow12**
## What did you implement:

- Updated parameter name in APIM arm template
- Fixed bug of sorting deployments in ascending order, when it should have been descending. This would have pretty serious consequences, because it means that the comparison of ARM templates would always be targeting the first ever deployment, not the most recent.
- Because the `sort()` function sorts the array in place, this bug was not being detected by the tests. Updated `resourceService` tests to create copies of the array rather than using the original reference when testing the validity of the result.

## How did you implement it:

- Flipped the timestamp comparison operator
- Updated the string in `apim.ts`

## How can we verify it:

- Deploy a service
- Make changes to an environment variable
- Deploy it again (should deploy ARM template)
- Change back to the original configuration
- Deploy it again (should deploy ARM template)

Without this fix, the last step would not have deployed the ARM template since it would be identical to the first deployment ever.

## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

- [x] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [x] Write documentation
- [x] Ensure there are no lint errors.  
       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: src/services/resourceService.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export class ResourceService extends BaseService {
     this.log(`Listing deployments for resource group '${this.resourceGroup}':`);
     const deployments = await this.resourceClient.deployments.listByResourceGroup(this.resourceGroup);
     return deployments.sort((a: DeploymentExtended, b: DeploymentExtended) => {
-      return (a.properties.timestamp > b.properties.timestamp) ? 1 : -1
+      return (a.properties.timestamp < b.properties.timestamp) ? 1 : -1
```
Comment (tbarlow12) [inline src/services/resourceService.ts]: 😨

File: src/services/resourceService.test.ts [hunk 2]
```diff
@@ -102,10 +105,10 @@ describe("Resource Service", () => {
     const service = new ResourceService(sls, options);
     const deploymentString = await service.listDeployments();
     let expectedDeploymentString = "\n\nDeployments";
-    const originalTimestamp = +MockFactory.createTestTimestamp();
+    const originalTimestamp = +MockFactory.createTestTimestamp() + 4;
```
Comment (mydiemho) [inline src/services/resourceService.test.ts]: What's the `+` in front of MockFactory?

File: src/services/resourceService.test.ts [hunk 3]
```diff
@@ -102,10 +105,10 @@ describe("Resource Service", () => {
...
     let i = 0
-    for (const dep of deployments) {
-      const timestamp = originalTimestamp + i
+    for (const dep of [...deployments].reverse()) {
```
Comment (mydiemho) [inline src/services/resourceService.test.ts]: how is this being reverse? what's the comparison operator?

File: src/services/resourceService.test.ts (hunk 2 above)
Comment (tbarlow12) [reply src/services/resourceService.test.ts]: Parse as a number

File: src/services/resourceService.test.ts (hunk 3 above)
Comment (tbarlow12) [reply src/services/resourceService.test.ts]: The original `deployments` object is in ascending order by timestamp. We want to make sure the deployments come back in descending order by timestamp.

The comparison operator isn't in the tests, it's in the actual code - `resourceService` line 25

File: src/services/resourceService.test.ts (hunk 2 above)
Comment (mydiemho) [reply src/services/resourceService.test.ts]: magic :p

File: src/armTemplates/resources/apim.ts [hunk 4]
```diff
@@ -61,7 +61,7 @@ export class ApimResource implements ArmResourceTemplateGenerator {
           "location": "[parameters('location')]",
           "sku": {
             "name": "[parameters('apimSkuName')]",
-            "capacity": "[parameters('apimCapacity')]"
+            "capacity": "[parameters('apimSkuCapacity')]"
```
Comment (wbreza) [inline src/armTemplates/resources/apim.ts]: Nice catch!

File: src/services/resourceService.test.ts (hunk 2 above)
Comment (wbreza) [reply src/services/resourceService.test.ts]: woah - really?? Is that a new typescript thing?

Comment (wbreza) [inline src/services/resourceService.test.ts]: Nit: Missing semicolon at end of line.

File: src/services/resourceService.test.ts [hunk 5]
```diff
@@ -102,10 +105,10 @@ describe("Resource Service", () => {
...
+      const timestamp = originalTimestamp - i
```
Comment (wbreza) [inline src/services/resourceService.test.ts]: Nit: Missing semicolon at end of line.

**Pull request merged**